src/react-native/xcode.ts

```diff
@@ -142,17 +157,73 @@ export function addSentryWithBundledScriptsToBundleShellScript(
   if (isLikelyExpoScript) {
     const SENTRY_REACT_NATIVE_XCODE_PATH =
       "`\"$NODE_BINARY\" --print \"require('path').dirname(require.resolve('@sentry/react-native/package.json')) + '/scripts/sentry-xcode.sh'\"`";
-    return script.replace(
+    patchedScript = script.replace(
       /^.*?(packager|scripts)\/react-native-xcode\.sh\s*(\\'\\\\")?/m,
       // eslint-disable-next-line no-useless-escape
       (match: string) => `/bin/sh ${SENTRY_REACT_NATIVE_XCODE_PATH} ${match}`,
     );
   }
 
-  return script;
+  if (patchedScript === script) {
+    // No changes were made
+    clack.log.error(
+      `Failed to patch ${chalk.cyan(
+        'Bundle React Native code and images',
+      )} build phase.`,
+    );
+    if (isLikelyExpoScript) {
+      return new ErrorPatchSnippet(
+        makeCodeSnippet(true, (unchanged, plus, _minus) => {
+          return unchanged(
+            `${plus(
+              `/bin/sh \`"$NODE_BINARY" --print "require('path').dirname(require.resolve('@sentry/react-native/package.json')) + '/scripts/sentry-xcode.sh'"\``,
+            )} \`"$NODE_BINARY" --print "require('path').dirname(require.resolve('react-native/package.json')) + '/scripts/react-native-xcode.sh'"\``,
+          );
+        }),
+      );
+    } else {
+      // plain react-native
+      return new ErrorPatchSnippet(
+        makeCodeSnippet(true, (unchanged, plus, _minus) => {
+          return unchanged(`WITH_ENVIRONMENT="$REACT_NATIVE_PATH/scripts/xcode/with-environment.sh"
+REACT_NATIVE_XCODE="$REACT_NATIVE_PATH/scripts/react-native-xcode.sh"
+
+/bin/sh -c "$WITH_ENVIRONMENT ${plus(
+            `\\"/bin/sh ../node_modules/@sentry/react-native/scripts/sentry-xcode.sh `,
+          )}$REACT_NATIVE_XCODE${plus(`\\"`)}"
+`);
+        }),
+      );
+    }
+  }
+
+  return patchedScript;
 }
 
-export function addSentryWithCliToBundleShellScript(script: string): string {
+export function addSentryWithCliToBundleShellScript(
+  script: string,
+): string | ErrorPatchSnippet {
+  if (!script.includes('$REACT_NATIVE_XCODE')) {
+    clack.log.error(
+      `Could not find $REACT_NATIVE_XCODE in ${chalk.cyan(
+        'Bundle React Native code and images',
+      )} build phase. Skipping patching.`,
+    );
+    return new ErrorPatchSnippet(
+      makeCodeSnippet(true, (unchanged, plus, _minus) => {
+        return unchanged(`${plus(`export SENTRY_PROPERTIES=sentry.properties
+export EXTRA_PACKAGER_ARGS="--sourcemap-output $DERIVED_FILE_DIR/main.jsbundle.map"
+`)}
+/bin/sh -c "$WITH_ENVIRONMENT ${plus(
+          `\\"../node_modules/@sentry/cli/bin/sentry-cli react-native xcode`,
+        )} $REACT_NATIVE_XCODE${plus(`\\"`)}"
+${plus(
+  `/bin/sh -c "$WITH_ENVIRONMENT ../node_modules/@sentry/react-native/scripts/collect-modules.sh`,
+)}"
```

Comment:
The last " was not marked as added.

<img width="1017" alt="Screenshot 2025-04-11 at 18 56 59" src="https://github.com/user-attachments/assets/dc1ee453-8cd8-4f0b-b2e8-83698f88a1bd" />

````suggestion
  `/bin/sh -c "$WITH_ENVIRONMENT ../node_modules/@sentry/react-native/scripts/collect-modules.sh"`,
)}
````